src/lib/ERC7498NFTRedeemables.sol

```diff
@@ -430,18 +452,31 @@ contract ERC7498NFTRedeemables is IERC7498, DynamicTraits, RedeemablesErrors {
                     uint256 newTraitValue = uint256(currentTraitValue) - uint256(traitValue);
 
                     // Decrement the trait by the trait value.
-                    _setTrait(traitRedemptions[i].traitKey, traitRedemptions[i].identifier, bytes32(newTraitValue));
+                    DynamicTraits(token).setTrait(
+                        traitRedemptions[i].identifier, traitRedemptions[i].traitKey, bytes32(newTraitValue)
+                    );
+                } else if (substandard == 4) {
+                    // Revert if the current trait value is not equal to the substandard value.
+                    if (currentTraitValue != substandardValue) {
+                        revert InvalidRequiredValue(currentTraitValue, substandardValue);
```

Comment:
maybe this error should be InvalidRequiredTraitValue and include the traitKey and tokenId and token address as well, to be explicit about how to debug it